### DISCUSSION TITLE
Publish as a package

### DISCUSSION THREAD

**Discussion by eyalcohen4** [steven-tey/novel#53] [Ideas]
Hey! I would love to use novel as part of a project I'm working on.
What do you think about releasing the `Editor` component as an NPM package?

I would love to help with that if decided to go with it.

**Comment by steven-tey** [comment]
@eyalcohen4 perfect timing – was actually planning to add that next!

Would definitely love your help with this – feel free to kick a PR off and we can iterate on it until it's ready to be merged/published to NPM! 

P.S.: I've reached out to GitHub support to get the `novel` package on NPM – feel free to use something like `novel-editor` as a placeholder for now!

---

Proposed API design (WIP):

```
import { Editor } from `novel`

<Editor
	onUpdate={} // emitted on each update event
	onDebouncedUpdate={} // emitted on each debounced update event
	debounceDuration={} // in ms, default is 750 ms
	extensions={} // add more extensions or overwrite existing ones
	editorProps={} // add more editorProps or overwrite existing ones
/> 
```

**Comment by eyalcohen4** [comment]
Awesome! starting to work on that. 
wdyt about using turbo with `apps/*` and `packages/*` structure? 
I'm starting from there but we can also do something simpler

**Reply by steven-tey** [reply]
@eyalcohen4 that sounds perfect! :)

**Reply by eyalcohen4** [reply]
perfect tnx! made some progress here: https://github.com/eyalcohen4/novel 
will have a PR this week :)

**Reply by steven-tey** [reply]
@eyalcohen4 awesome – can't wait! :star-struck_cat:

**Reply by junoriosity** [reply]
@eyalcohen4 @steven-tey Is there some update to this? :)

**Reply by eyalcohen4** [reply]
The PR is up here: https://github.com/steven-tey/novel/pull/87
Have few tasks to close before we can publish but it's pretty close

**Reply by steven-tey** [reply]
🥳 This official `novel` SDK is now live: https://github.com/steven-tey/novel/discussions/124
